src/device/wg_controller.rs

```diff
@@ -841,12 +841,24 @@ async fn select_endpoint_for_peer<'a>(
     actual_peer: &Option<Peer>,
     time_since_last_rx: &Option<Duration>,
     peer_state: PeerState,
-    checked_endpoint: &Option<WireGuardEndpointCandidateChangeEvent>,
+    mut checked_endpoint: &Option<WireGuardEndpointCandidateChangeEvent>,
     proxy_endpoint: &[SocketAddr],
     upgrade_request_endpoint: &Option<SocketAddr>,
 ) -> Result<(Option<SocketAddr>, Option<EndpointState>)> {
     // Retrieve some helper information
-    let actual_endpoint = actual_peer.clone().and_then(|p| p.endpoint);
+    let actual_endpoint = actual_peer.as_ref().and_then(|p| p.endpoint);
+
+    let is_actual_endpoint_newer = {
+        let changed_at = actual_peer.as_ref().and_then(|p| p.endpoint_changed_at);
+        let checked_at = checked_endpoint.as_ref().map(|e| e.changed_at);
+        match (changed_at, checked_at) {
+            (Some(wg), Some(cpc)) => wg >= cpc,
```

Comment:
I just realized one edge-case, which is a bit sus. Therefore would like to discuss this.

Lets imagine, that for some reason, telio-proxy decides to change it's mapping and thus relay->relay endpoint change occurs. I think this will not trigger endpoints to be invalidated in device/mod.rs, meaning that CPC will not attempt to publish any endoints, but this conditions will prevent to use them for the forseeable future.

Reply:
To fix it, in we may add this condition, that in case CPC is holding checked endpoints, which are older than currently used relay endpoint -> we must invalidate them. 

This may replace existing `was_proxing == false && is_proxying == true` condition. As the one based on `changed_at` is more general and independant of execution cycle :+1:

Reply:
It would be very rare, the only reason for proxy to change is socket failure.

The wailure scenario would be that we would stop not do endpoint discovery for some time?

Reply:
Yeh, I think it's fair, we will be independent of other places

Reply:
> The wailure scenario would be that we would stop not do endpoint discovery for some time?

Failure scenario would be that no connection upgrades to direct would occur until restart or network change.